utils/orchestrators.ts

```diff
@@ -0,0 +1,120 @@
+import * as df from "durable-functions";
+import { DurableOrchestrationClient } from "durable-functions/lib/src/durableorchestrationclient";
+import { array } from "fp-ts/lib/Array";
+
+import { toError } from "fp-ts/lib/Either";
+import {
+  fromLeft,
+  taskEither,
+  TaskEither,
+  tryCatch
+} from "fp-ts/lib/TaskEither";
+import {
+  IResponseErrorConflict,
+  IResponseErrorInternal,
+  IResponseSuccessAccepted,
+  ResponseErrorConflict,
+  ResponseErrorInternal,
+  ResponseSuccessAccepted
+} from "italia-ts-commons/lib/responses";
+
+import { FiscalCode } from "italia-ts-commons/lib/strings";
+import { PromiseType } from "italia-ts-commons/lib/types";
+import { StatusEnum as CgnActivatedStatusEnum } from "../generated/definitions/CgnActivatedStatus";
+import { StatusEnum as CgnCanceledStatusEnum } from "../generated/definitions/CgnCanceledStatus";
+import { StatusEnum as CgnPendingStatusEnum } from "../generated/definitions/CgnPendingStatus";
+import { StatusEnum as CgnRevokedStatusEnum } from "../generated/definitions/CgnRevokedStatus";
+import { CgnStatus } from "../generated/definitions/CgnStatus";
+
+/**
+ * The identifier for StartEligibilityCheckOrchestrator
+ * @param fiscalCode the id of the requesting user
+ */
+export const makeUpdateCgnOrchestratorId = (
+  fiscalCode: FiscalCode,
+  cgnStatus: string
+) => `${fiscalCode}-UPDCGN-${cgnStatus}`;
+
+/**
+ * Returns the status of the orchestrator augmented with an isRunning attribute
+ */
+export const isOrchestratorRunning = (
+  client: DurableOrchestrationClient,
+  orchestratorId: string
+): TaskEither<
+  Error,
+  PromiseType<ReturnType<typeof client["getStatus"]>> & {
+    isRunning: boolean;
+  }
+> =>
+  tryCatch(() => client.getStatus(orchestratorId), toError).map(status => ({
+    ...status,
+    isRunning:
+      status.runtimeStatus === df.OrchestrationRuntimeStatus.Running ||
+      status.runtimeStatus === df.OrchestrationRuntimeStatus.Pending
+  }));
+
+const cgnStatuses: ReadonlyArray<string> = [
+  CgnRevokedStatusEnum.REVOKED.toString(),
+  CgnActivatedStatusEnum.ACTIVATED.toString(),
+  CgnCanceledStatusEnum.CANCELED.toString(),
+  CgnPendingStatusEnum.PENDING.toString()
+];
+
+type CheckUpdateCgnIsRunningErrorTypes =
+  | IResponseErrorInternal
+  | IResponseSuccessAccepted
+  | IResponseErrorConflict;
+/**
+ * Check if the current user has a pending cgn status update process.
+ */
+export const checkUpdateCgnIsRunning = (
+  client: DurableOrchestrationClient,
+  fiscalCode: FiscalCode,
+  cgnStatus: CgnStatus
+): TaskEither<CheckUpdateCgnIsRunningErrorTypes, false> =>
```

Comment:
I'd rather not map Http responses here, and let the caller do the eventual mapping. Instead, I would provide some orchestrator-related infos: maybe a string union will be enough.

Reply:
By doing this we lost error messages, so the caller can't log exactly what's the matter.  What do you think @balanza ?